### ISSUE TITLE
Interval.splitAt() returns invalid Intervals when DateTime is not within Interval

### ISSUE THREAD

**Issue by bw** [moment/luxon#553]
The `Interval.splitAt()` function returns incorrect values when one of the provided DateTimes (to split at) is before the Interval.

This behavior does not occur when the provided DateTime is _after_ the Interval, so I believe this is a bug.

To reproduce, consider the following code:

```js
const testInterval = Interval.fromDateTimes(
  DateTime.fromMillis(1565712900000),
  DateTime.fromMillis(1565928000000),
);
const testDT = DateTime.fromMillis(1565647200000);

const splitIntervals = testInterval.splitAt(testDT);
```

The expected return value is (more or less) the original Interval in an array (i.e. `[testInterval]`). Instead, we get two Intervals, the first of which is invalid because the end must be after the start -- and the second of which is just plain wrong (returns Interval that has start `testDT` and end `testInterval.end`).

I'm not sure if we should expect this to work but other functions do not silently fail when invalid values are provided.

A temporary workaround would be to check that provided DTs are within the Interval range before passing them to `splitAt`.

**Comment by icambron** [comment]
Yeah, I agree. I'm on the fence on whether to make before just return the original or simply throw an exception. But either way, yes, we should fix this.

**Comment by bw** [comment]
Thanks for the reply!

IMO since the documentation defines this function as "Split this Interval at each of the specified DateTimes", the behavior should be just to return the original Interval in an Array. I think of this for two reasons--

First, throwing an exception would require end users like myself to catch and consider all the possible edge cases. The circumstances under which a Luxon function throws an exception would have to be well documented. This would be difficult from a documentation perspective.

Second, there is almost no way this (exception throwing) behavior could be consistent across Luxon functions. For example, if you pass a Duration to `Interval.splitBy()` that is longer than the Interval, should it also throw an Exception? Shouldn't `Interval.intersection()` throw an exception instead of returning `null` for non-overlapping intervals?

Just my 2c :)

**Comment by icambron** [comment]
Fixed in 1.19.0
